R/shiny.R

```diff
@@ -555,15 +555,16 @@ dataTableAjax = function(session, data, rownames, filter = dataTablesFilter, out
   data = as.data.frame(data)  # think dplyr
   if (length(rn)) data = cbind(' ' = rn, data)
 
-  sessionDataURL(session, data, outputId, filter)
+  sessionDataURL(session, data, outputId, filter, future)
 }
 
-sessionDataURL = function(session, data, id, filter) {
+#' @importFrom promises future_promise
+sessionDataURL = function(session, data, id, filter, future) {
 
   toJSON = shinyFun('toJSON')
   httpResponse = shinyFun('httpResponse')
 
-  filterFun = function(data, req) {
+  filterExpr = expression({
```

Comment:
Usually I don't prefer using `eval(expression())` because we need to be careful about the evaluating environment. Is it okay to still make a function here, and provide a function call in `promises::future_promise()`?

Reply:
```suggestion
  filterFun2 = function(data, req) {
```

Reply:
Sounds good to me